src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -75,14 +81,30 @@ pub enum AbstractDataType {
     Structure(bool),
 }
 
+impl AbstractDataType {
+    pub fn element_type(&self) -> Option<String> {
```

Comment:
In methods throughout: it's more conventional to return an `Option<&str>` and let the end user decide whether to call `to_owned()` on it. (Doesn't need to happen in this PR.)

Reply:
Yes. I though of using `&str` but then its used in places where `String` is required, so it was either having `to_owned` after this method or just returning an owned value form here.